packages/plugin-compiler/src/main/kotlin/io/realm/kotlin/compiler/RealmModelSyntheticPropertiesGeneration.kt

```diff
@@ -367,6 +383,9 @@ class RealmModelSyntheticPropertiesGeneration(private val pluginContext: IrPlugi
         val fields: MutableMap<String, SchemaProperty> =
             SchemaCollector.properties.getOrDefault(irClass, mutableMapOf())
 
+        // A map for tracking the names (and their locations) used to ensure uniqueness
+        val internalAndPublicNames = mutableMapOf<String, CompilerMessageSourceLocation>()
```

Comment:
#naming The name of this property is a bit vague.... but not sure I have an obvious better name... but this feels more like a `internalNameToSourceLocation` kind of thing...that one is also a mindful though

Reply:
Agreed, suggestions:
* `persistedAndPublicNamesToLocations`
* `persistedAndPublicNameToLocation`
* `persistedAndPublicNameLocations`
(or replace `Location` with `SourceLocation`)

Reply:
IMO `persistedAndPublicNameToLocation` would be more descriptive than the current one.